### PR TITLE
[vmap] implement batching rules for clamp, clamp_min and clamp_max

### LINKED ISSUE #47754: Add batching rule for clamp, clamp_min, clamp_max
## 🚀 Feature

Support efficient computation of clamp, clamp_min, and clamp_max inside of vmap.

## To Reproduce
```
>>> import torch
>>> from torch import vmap
>>> x = torch.randn(64, 3)
>>> result = vmap(lambda x: x.clamp(min=0))(x)
UserWarning: Batching rule not implemented for aten::clamp falling back to slow (for loop and stack) implementation (Triggered internally at  ../aten/src/ATen/BatchedFallback.cpp:68.)
```

After this task is complete, we should not see the user warning anymore. Ditto for clamp_min and clamp_max.

## Context

[`vmap` is a higher-order vectorization operator](https://pytorch.org/docs/master/generated/torch.vmap.html). It takes in a function `f` and returns a new function that maps the `f` over some dimension of the inputs. The new function has similar semantics to performing a for-loop over the mapped dimension:
```py
inputs = torch.randn(N, 3)
f = lambda x: x.clamp(min=0)
expected = torch.stack([f(inputs[i]) for i in range(N)])
result = torch.vmap(f)(inputs)
assert torch.allclose(result, expected)
```
Instead of performing a for-loop to compute `expected`, the user could have vectorized their computation by straight up calling `clamp`: `expected2 = inputs.clamp(min=0)`. However, for more complicated functions, like `F.conv2d`, vectorizing is not as obvious. In some cases, like efficient Jacobian/Hessian computation, vectorization isn't possible by the user because they cannot control what happens inside of the autograd engine. One of the targeted use cases for vmap is to speed up batched gradient computation in PyTorch.

How vmap works is that a "batching rule" is defined for each operator. The batching rule defines how to perform the operation on inputs with extra dimensions (that we refer to as the "batch dimensions").

## What you should do

Because clamp is a simple operation, its "batching rule" should just be to call clamp on the original tensor:
- `expected = torch.stack([f(inputs[i]) for i in range(N)])` and `expected = f(inputs)` produce the same result (for `f = lambda x: x.clamp(min=0)`. Much of the complexity of the task is navigating the abstractions we use to implement batching rules.

Concretely, the task is to implement a BatchedTensor override for the following operators:
- clamp
- clamp_min
- clamp_max

## Resources 

- [How to write a batching rule](https://github.com/pytorch/pytorch/blob/635aebdfab20d9ace8ecfaf347102f0a49fa3553/aten/src/ATen/BatchingRegistrations.cpp#L8-L49)
- [What is a BatchedTensorImpl](https://github.com/pytorch/pytorch/blob/635aebdfab20d9ace8ecfaf347102f0a49fa3553/aten/src/ATen/BatchedTensorImpl.h#L45-L55)
- [What is](https://github.com/pytorch/pytorch/blob/635aebdfab20d9ace8ecfaf347102f0a49fa3553/aten/src/ATen/VmapTransforms.h#L7-L26) [a vmap transform](https://github.com/pytorch/pytorch/blob/635aebdfab20d9ace8ecfaf347102f0a49fa3553/aten/src/ATen/VmapTransforms.h#L41-L47)
- [Sample batching rule implementation](
https://github.com/pytorch/pytorch/blob/635aebdfab20d9ace8ecfaf347102f0a49fa3553/aten/src/ATen/BatchingRegistrations.cpp#L210-L216)

### PR DESCRIPTION
**Pull request by RockingJavaBean**
Fix #47754

- This PR implements batching rules for `clamp`, `clamp_min` and `clamp_max` operators.
- Testcases are added to `test/test_vmap.py`.

### CODE REVIEW

**Review by zou3519 — APPROVED** [review]
thanks!
